File: web/plugins/string_length.js
import { WebPlugin } from "../engine/plugin.js";

export class StringLengthPlugin extends WebPlugin {
    constructor() {
        super("文字の長さ");
    }

    async execute(line, context) {
        const match = line.match(/^文字の長さ\s+「(.+)」$/);
        if (!match) return;

        context.runtime.write(match[1].length);
    }
}

File: pc/plugins/string_length.py
import re
from core.plugin import Plugin


class StringLengthPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'文字の長さ\s+「(.+)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print(len({value!r}))"